cucumber/vscode
# Remove unused gherkin grammar for syntax highlight

### 🤔 What's changed?

Removed unused grammar file for classic gherkin.

### ⚡️ What's your motivation? 

- Source of confusion - easy to make [assumption is in use](#153).
- General tidy - removal of dead code - syntax highlighting is implemented through the [Semantic Token Provider](https://code.visualstudio.com/api/references/vscode-api#DocumentSemanticTokensProvider) in the [Language Service](https://github.com/cucumber/language-service/blob/1e34a962320b9632ddf3bac04e46ea2806183250/src/service/getGherkinSemanticTokens.ts). Grammar file appears to have been [introduced when creating the repository intially](https://github.com/cucumber/vscode/commit/9a2258aa3087aa6e7f565f72c4d8aba3176de2d9).
- Remains accessible through the archived [cucumber-attic/cucumber-tmbundle repository](https://github.com/cucumber-attic/cucumber-tmbundle/blob/17a4419fb5ea47f910d9b2fd8b29f01f68382050/Syntaxes/Cucumber%20Plain%20Text%20Feature.tmLanguage)

To verify is unused:

- Open the extension and use the `Rule:` keyword; it will highlight. As this keyword is not part of the earlier TextMate-defined grammar, you will not find a reference to it in that file - unlike other keywords.
- Check `package.json`. The `grammars` contribution endpoint is not set, like so (see [VSCode docs](https://code.visualstudio.com/api/language-extensions/syntax-highlight-guide#contributing-a-basic-grammar)):

    ```json
    {
      "contributes": {
        "grammars": [
          {
            "language": "abc",
            "scopeName": "source.abc",
            "path": "./syntaxes/abc.tmGrammar.json"
          }
        ]
      }
    }
    ```

- Run the contributing [Get Started](https://github.com/cucumber/vscode/blob/8a257f1635fa7ec4be442f278e639961d1016a21/CONTRIBUTING.md) guidelines with the `syntaxes` directory removed; syntax highlighting remains working.

### 🏷️ What kind of change is this?

- :bank: Refactoring/debt/DX (improvement to code design, tooling, documentation etc. without changing behaviour)

### ♻️ Anything particular you want feedback on?



### 📋 Checklist:

- [x] I agree to respect and uphold the [Cucumber Community Code of Conduct](https://cucumber.io/conduct/)
